scripts/merge-script/cli.ts

```diff
@@ -0,0 +1,135 @@
+
+  // Merge the pull request and handle the result. If failures have been reported, exit
+  // the process with a non-zero exit code.
+  if (!await handleMergeResult(await api.merge(pullRequests, dryRun, false), force)) {
+    process.exit(1);
+  }
+
+  /**
+   * Prompts whether failed pull requests should be forcibly merged. If so, merges
+   * the specified failed pull requests forcibly (ignoring non-critical failures).
+   * @returns Whether the failed pull requests have been forcibly merged.
+   */
+  async function promptAndPerformForceMerge(failedPrs: number[]): Promise<boolean> {
+    if (await promptConfirm('Do you want to forcibly proceed with merging?')) {
+      // Perform a merge in force mode and handle the merge result. Note that we disable the
+      // force merge prompt since a failed force merge will never succeed with a second force merge.
+      return await handleMergeResult(await api.merge(failedPrs, dryRun, true), true);
+    }
+    return false;
+  }
+
+  /**
+   * Handles the merge result by printing console messages, exiting the process
+   * based on the result, or by restarting the merge if force mode has been enabled.
+   * @returns Whether the merge was successful or not.
+   */
+  async function handleMergeResult(
+      {failures, status, merged}: MergeResult, disableForceMergePrompt = false): Promise<boolean> {
+    const canForciblyMerge = failures ?
+        failures.every(failure => FORCE_OVERRIDABLE_FAILURES.includes(failure.status)) :
+        false;
+
+    switch (status) {
+      case MergeStatus.SUCCESS:
+        console.info(chalk.green('Successfully merged all pull requests.'));
+        printMergedPullRequests(merged!);
+        return true;
+      case MergeStatus.DIRTY_WORKING_DIR:
+        console.error(chalk.red(
+            `Local working repository not clean. Please make sure there are ` +
+            `no uncommitted changes.`));
+        return false;
+      case MergeStatus.UNKNOWN_GIT_ERROR:
+        console.error(chalk.red(
+            'An unknown Git error has been thrown. Please check the output ' +
+            'above for details.'));
+        return false;
+      case MergeStatus.FAILED:
+        console.error(chalk.red(
+            `Could not merge the specified pull requests. The following ` +
+            `errors need to be resolved:`));
+        printPullRequestsWithErrors(failures!);
+        if (canForciblyMerge && !disableForceMergePrompt) {
+          console.info();
+          console.info(chalk.yellow('The pull requests above failed due to non-critical errors.'));
+          console.info(chalk.yellow(`These errors can be forcibly ignored if desired.`));
+          return await promptAndPerformForceMerge(failures!.map(f => f.prNumber));
+        }
+        return false;
+      default:
+        throw Error(`Unexpected merge result: ${status}`);
+    }
+  }
+}
+
+/** Parses the command line and returns the passed options. */
+function parseCommandLine(): {
```

Comment:
Should we use some kind of command line parsing library? I'm always tempted to write stuff like this in Python since they have really nice parsing built in

Reply:
Yeah agreed! The plan is to use `yargs` for this if we move it over to `dev-infra/` in `angular/angular`. We decided to do that as a follow-up.

Reply:
Add a TODO?